File: Euler1/Euler1.rs
// find the sum of all multiples of 3 or 5 between 1 and 1000
use std::time::Instant;

fn main() {
    let start = Instant::now();
    let mut bar = 0;
    let mut sum = 0;
    while bar < 1000 {
        sum += bar;
        bar += std::cmp::min(3 - (bar % 3), 5 - (bar % 5));
    }
    println!("Time\t|\t{}e-6 s", start.elapsed().subsec_micros());
    println!("--------------------------");
    println!("Sum\t|\t{}", sum);
}

File: Euler1/Euler1.clj
;; find the sum of all multiples of 3 or 5 from 0 to 10000
(defn applies [x]
  (or (= 0 (mod x 3))
      (= 0 (mod x 5))
      ))

(pr (reduce + (filter applies (take 999 (map #(+ 1 %) (range))))))

